Orange/base.py

```diff
@@ -27,11 +27,15 @@ def __init__(self, preprocessors=None):
         self.preprocessors = list(preprocessors)
 
     def fit(self, X, Y, W=None):
-        raise NotImplementedError(
-            "Descendants of Learner must overload method fit")
+        raise SystemError(
```

Comment:
Neither is appropriate. `NotImplemented` is described as

> In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method.

We do not require overriding `fit`; derived class may provide `fit_storage` instead. Lint doesn't know that, so if we use `NotImplementedError`, lint complains about undefined abstract method in derived class. I used `SystemError` in the sense "this shouldn't have happened". But you're right, it wasn't meant for this. I'll go for [RuntimeError](https://docs.python.org/3.5/library/exceptions.html#RuntimeError).